### PR TITLE
Extract browserify code

### PR DESCRIPTION
**Pull request by craigbeck**

### CODE REVIEW

File: lib/bundler/index.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+// Avoid Browserifying these dependencies
+let crypto, fs, path, through;
```
Comment (ericyhwang) [inline lib/bundler/index.js:2]: Let's stick to `var` for now here and elsewhere in this file, for consistency with the rest of the Derby code, until we get around to converting it to ES6+ syntax.

File: lib/bundler/index.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+
+module.exports = function derbyBundler(app, options) {
+  if (process.title === 'browser') {
+    // not in server
+    return;
+  }
+  const { App, util } = app.derby;
+
+  App.prototype.bundle = function(backend, options, cb) {
```
Comment (ericyhwang) [inline lib/bundler/index.js:18]: Would you happen to know if this is sufficient to keep this code out of the client JS bundle?

Currently the `*ForServer` files are dynamically required, which I believe keeps them out of the client bundle entirely:
https://github.com/derbyjs/derby/blob/4d3327b4c389d9f763ac40baf8e940f8ad901c02/index.js#L2

I'm not sure if this by itself is going to do the same. Might be on whoever requires this `lib/bundler` file to guard appropriately? Or maybe the bit of extra code isn't a big enough deal to matter?

Comment (ericyhwang) [reply lib/bundler/index.js:18]: Ah, if this file only gets required from server JS files, then that's fine.

I like the idea of a thrown exception explaining that this should only be required from server code, reduces the possibility of accidentally including it.

File: lib/bundler/index.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  through = require('through');
+}
+
+
+module.exports = function derbyBundler(app, options) {
+  if (process.title === 'browser') {
+    // not in server
+    return;
+  }
+  const { App, util } = app.derby;
```
Comment (ericyhwang) [inline lib/bundler/index.js:16]: I see a few other uses of `const` and let` in the file, to switch back to `var` for now.

Comment (craigbeck) [reply lib/bundler/index.js:16]: fixed them all

**Review by ericyhwang — APPROVED** [review]

**Pull request merged**